### PR TITLE
Add (kinda gross) tab based auto-complete

### PR DESCRIPTION
**Pull request by rolandshoemaker**
Adds tab autocomplete + listing possible actions and also history (up key for prev actions). I haven't written python in a really long time so totally relying on you checking this Noah. It totally worked for all the things I tried!

### CODE REVIEW

**Comment by SwartzCr** [comment]
siiiick - I'll look it over when I get some spare cycles
BUT FOR NOW I AM EXCITED

File: lib/display.py [hunk 1]
```diff
@@ -136,6 +138,25 @@ def quit(data):
...
+class actionCompleter(object):
+
+    def __init__(self):
+        return
+
+    def set_actions(self, actions):
+        self.actions = sorted(actions)
+
+    def complete(self, action, index):
+        if index == 0:
```
Comment (SwartzCr) [inline lib/display.py:152]: can you explain this line to me?

**Comment by SwartzCr** [comment]
It doesn't seem to work on commands with spaces - any idea why this would be? I only kind of understand your code...
feel free to try to explain it to me
Also in the situation where it won't complete on a space (e.g. 'check' in the initial screen) it will still list all options rather than just the ones that start with check when you hit tab.

File: lib/display.py (hunk 1 above)
Comment (rolandshoemaker) [inline lib/display.py:152]: `complete` is called multiple times per input `action` if the previous call to it returned a string, in this case `index` is incremented, if `None` is returned it assumes all possible completions have been returned and goes on its way. We only build the match list the first iteration to save a bit of time.

**Comment by rolandshoemaker** [comment]
Whoops, `readline` will only pass the current word to `complete` (very weird) so it needs to store some internal state to properly match against multi-word actions (i.e. only offer the second halves of actions). I'll take a crack at fixing this tonight!

**Comment by rolandshoemaker** [comment]
I think I hit all the corner cases on this, `readline` is a bit strange.

**Comment by SwartzCr** [comment]
It seems to be working - except it'd be nice if when you were tab completeing on two word commands it'd still print the 2 word version but fill just the remaining word - though i'm too groggy to figure out how to do that since some of this is still magic to me
So I think I'll merge and then have that be a future feature

**Pull request merged**